domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt

```diff
@@ -95,15 +106,36 @@ class TranslationController @Inject constructor(
     }
   }
 
+  /**
+   * Returns List data for [SupportedLanguages] available language definitions from proto asset file.
+   */
+  fun getAllAppLanguageDefinitions(): List<OppiaLanguage> {
+    val supportedLanguages = languageConfigRetriever.loadSupportedLanguages()
+    val supportedLanguageList = arrayListOf<OppiaLanguage>()
+    for (LanguageSupportDefinition in supportedLanguages.languageDefinitionsList) {
+      if (LanguageSupportDefinition.hasAppStringId()) {
+        supportedLanguageList.add(LanguageSupportDefinition.language)
+      }
+    }
+    return supportedLanguageList
```

Comment:
```suggestion
    val supportedLanguages = languageConfigRetriever.loadSupportedLanguages()
    return supportedLanguages.languageDefinitionsList.filter { 
      it.hasAppStringId()
    }.map { it.language }
```

Bit more Kotlin-idiomatic.

Also, variable names should always be in lowerCamelCase, not UpperCamelCase (per your ``LanguageSupportDefinition`` variable).

Reply:
Thanks Ben makes sense. Updated this.